### PR TITLE
Se 519 create system level invariant tests for erc721

### PR DESCRIPTION
**Pull request by TJ-Everett**

### CODE REVIEW

File: docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md [hunk 1]
```diff
@@ -21,18 +21,18 @@
...
 - ownerOf() should revert if the token has been burned.
+- A burned token should have its approvals reset.
 
 ## [Mintable ERC721 Invariants](../../../../../../../test/client/token/ERC721/invariant/ApplicationERC721MintBurn.t.i.sol)
 - Mint increases the total supply
 - Mint creates a fresh applicationNFT
+- Minting tokens should update user's balance
 
 
 # Not Implemented
```
Comment (VoR0220) [inline docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md:32]: Let's get rid of this if these have now in fact been implemented.

File: test/client/token/ERC721/invariant/ApplicationERC721System.t.i.sol [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        selectors[0] = newAppManagerAddress.confirmAppManager.selector;
+        targetSelector(FuzzSelector({addr: address(newAppManagerAddress), selectors: selectors}));
+        vm.stopPrank();
+        vm.startPrank(USER2);
+        vm.expectRevert(abi.encodePacked("AccessControl: account ", Strings.toHexString(address(USER2)), " is missing role 0x371a0078bf8859908953848339bea5f1d5775487f6c2f50fd279fcc2cafd8c60"));
+        newAppManagerAddress.confirmAppManager(address(applicationNFT)); 
+    }
+
+    // When AppManagerAddress is confirmed, AppManagerAddressSet event is always emitted
+    function invariant_ERC721_external_ProposeAppManager_ConfirmAppManager_EventEmission() public {
```
Comment (Palmerg4) [inline test/client/token/ERC721/invariant/ApplicationERC721System.t.i.sol:213]: What is the reason this test logic is commented out?

**Review by Palmerg4 — CHANGES_REQUESTED** [review]
Noticed that the last test has logic commented out. Is this intentional?

File: test/client/token/ERC721/invariant/ApplicationERC721System.t.i.sol (hunk 2 above)
Comment (TJ-Everett) [reply test/client/token/ERC721/invariant/ApplicationERC721System.t.i.sol:213]: That's embarrassing. That was me trying to troubleshoot something and didn't go back to fix that test

File: docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md (hunk 1 above)
Comment (TJ-Everett) [reply docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md:32]: I thought this was the list of ones we have tested / want to test. Is the idea to remove them once tests are implemented?

**Review by ShaneDuncan602 — APPROVED** [review]
Great job. These invariant tests can be tricky. I love the use of excludeContracts and the dummyAcceptor contract.

**Review by Palmerg4 — APPROVED** [review]
Lgtm. Nice tests!

**Pull request merged**